contracts/KyberNetwork.sol

```diff
@@ -436,13 +436,20 @@ contract KyberNetwork is Withdrawable, Utils3, KyberNetworkInterface, Reentrancy
             searchBestRate(src, ETH_TOKEN_ADDRESS, srcAmount, usePermissionless);
 
         result.weiAmount = calcDestAmountWithDecimals(srcDecimals, ETH_DECIMALS, srcAmount, result.rateSrcToEth);
+        //if weiAmount is zero, return zero rate to avoid revert in ETH -> token call
+        if (result.weiAmount == 0) {
+            result.reserve2 = address(0);
+            result.rateEthToDest = 0;
+            result.destAmount = 0;
+            result.rate = 0;
```

Comment:
please write this in a way that created minimum code changes.
exapmle: if weiAmount is zero.
try to set result and return it. and leave rest of code untouched.

Reply:
Updated!